kyber/logs.py

```diff
@@ -0,0 +1,99 @@
+import click
+import heapq
+import iso8601
+import itertools
+import time
+
+from pykube.objects import Pod, ObjectDoesNotExist
+from kyber.lib.kube import kube_api
+
+
+class PriorityQueue(object):
+    """ Basic priority queue backed by a heap + a counter to tiebreak same-priority
+    values.
+    """
+    def __init__(self):
+        self.heap = []
+        self.counter = itertools.count()
+
+    @property
+    def count(self):
+        return len(self.heap)
+
+    @property
+    def empty(self):
+        return self.count == 0
+
+    def push(self, priority, value):
+        heapq.heappush(self.heap, (priority, next(self.counter), value))
+
+    def pop(self):
+        return heapq.heappop(self.heap)
+
+
+def parse_logentry(logentry):
+    """ Parse a log entry of fmt: `<iso8601> <log string>\n`
```

Comment:
Is this the format that kubectl provides for all pods or does this assume that the logs from the pod start with the iso8601 string?